src/api/MainSolver.cc

```diff
@@ -183,7 +188,41 @@ sstat MainSolver::simplifyFormulas() {
     return status;
 }
 
-void MainSolver::printFramesAsQuery(std::ostream & s) const {
+vec<PTRef> MainSolver::getCurrentAssertions() const {
+    vec<PTRef> assertions;
+    size_t const cnt = frames.frameCount();
+    for (size_t i = 0; i < cnt; ++i) {
+        for (PTRef fla : frames[i].formulas) {
+            assertions.push(fla);
+        }
+    }
+    return assertions;
+}
+
+vec<PTRef> const & MainSolver::getAssertionsAtLevel(std::size_t level) const {
+    assert(level >= 1);
+    assert(level <= getAssertionLevel());
+    return frames[level - 1].formulas;
+}
+
+void MainSolver::printCurrentAssertionsAsQuery() const {
+    char * base_name = config.dump_query_name();
+    if (base_name == NULL)
+        printCurrentAssertionsAsQuery(std::cout);
+    else {
+        char * s_file_name;
+        int chars_written = asprintf(&s_file_name, "%s-%d.smt2", base_name, check_called);
```

Comment:
There should be a C++ way to do this, no?
We can try `std::format` maybe?

Reply:
I did not care, I just renamed the old function but did not change the body. I think many parts like this will need to be updated.. Here I just focused on API.

Reply:
-> #788